src/main/java/sdmExam/Game.java

```diff
@@ -18,10 +18,21 @@ public void play(Stone player, Position position) throws Exception {
             throw new Exception("Position is already occupied.");
         }
 
+        if (isIllegalMove(player, position)) {
+            throw new Exception("A player cannot put a stone diagonally adjacent to another stone of the same colour" +
+                    " without a colour alike orthogonally adjacent.");
+        }
+
         board.addStoneAt(player, position);
         lastPlay = player;
     }
 
+    private boolean isIllegalMove(Stone player, Position position) {
+        final Intersection intersection = board.intersectionAt(position);
+        return board.existsDiagonallyAdjacentWithStone(intersection, player) &&
+                !board.existsOrthogonallyAdjacentWithStone(intersection, player);
+    }
+
```

Comment:
One question here, I am bit confused,think about the case player wants to put a stone to board (no orthogonal or diagonal adj like first move), with this method we don't allow this, do we? It may spawn a problem in first moves if it's the case.